### DISCUSSION TITLE
BIRT image iten rotates jpg images

### DISCUSSION THREAD

**Discussion by opabloc** [eclipse-birt/birt#1061] [Q&A] [answered]
On BIRT 4.3.2 image report item when is filled with any jpg image using URI,file in shared resources,embedded image or Dynamic image shows the image rotated to left. This doesn't happen with others formats like png or gif.

Thanks

**Comment by hvbtup** [comment, marked as answer]
Welcome to the wonderful world of image formats!

Probably the image is a photo?
Can you share an example image here?

We could then test if the issue still persists with the current BIRT release 4.9.

Some cameras or smartphones are "clever": If you tilt the device by 90° to the left or to the right°, they notice this fact.
The hardware/software which is creating a JPEG from the raw camera sensor data always works in the same way regardless of whether you are creating a portrait or landscape photo.
When saving the JPEG, information about what is actually "up" is written to the EXIF metadata (or other metadata).

New image processing software usually knows about this metadata and uses it to rotate the decompressed image before presenting it to the user.

Older image processing software - as is used in BIRT 4.3.2 - does not know about this metadata and thus shows a rotated image if the device was tilted by +/-90°.

This is explained in more detail eg. here: https://jdhao.github.io/2019/07/31/image_rotation_exif_info/

You will need to pre-process such images before showing them in BIRT.
